sdk/core/core-rest-pipeline/src/interfaces.ts

```diff
@@ -47,6 +47,39 @@ export interface HttpHeaders extends Iterable<[string, string]> {
   toJSON(options?: { preserveCase?: boolean }): RawHttpHeaders;
 }
 
+/**
+ * A part of the request body in a multipart request.
+ */
+export interface BodyPart {
+  /**
+   * The headers for this part of the multipart request.
+   */
+  headers: HttpHeaders;
```

Comment:
do we allow arbitary headers or limited set?

Reply:
Arbitrary headers are OK here, it's really up to the user what headers they want in each part. The form data policy just uses Content-Type and Content-Disposition but a different kind of multipart request could use different headers.